tools/reorder_docids.cpp

```diff
@@ -0,0 +1,12 @@
+#include <CLI/CLI.hpp>
+
+#include "app.hpp"
+#include "reorder_docids.hpp"
+
+int main(int argc, const char** argv)
+{
+    CLI::App app{"Reassigns the document IDs."};
+    pisa::ReorderDocuments args(&app);
+    CLI11_PARSE(app, argc, argv);
+    pisa::reorder_docids(args);
```

Comment:
cam we expand the arguments here?

```
pisa::reorder_docids(args.input_basename(),
                *args.output_basename(),
                args.document_lexicon(),
                args.reordered_document_lexicon(),
                args.seed()
                ...)
```

Reply:
There is a reason why it's not expanded. Mainly, it's because it's used in two binaries, which means you have to expand twice, and it's easy to make a mistake and make these two places diverge.

I suggest moving the specific functions like `reorder_bp` to the library, but keep that one function in `tools` so that it can be reused between `reorder_docids` and `shards`.

Reply:
I agree. Thanks